lib/Trash/PageTrashBackend.php

```diff
@@ -539,7 +539,8 @@ public function getTrashItemByCollectiveAndId(IUser $user, int $collectiveId, in
 			$trashNode = $this->getTrashNodeById($user, $fileId);
 			// Get parent folder for index pages
 			if ($trashNode instanceof File && NodeHelper::isIndexPage($trashNode)) {
-				$trashNode = $trashNode->getParent();
+				// The extra `get()` is required to resolve the lazy folder from getParent()
+				$trashNode = $trashNode->getParent()->get('');
```

Comment:
While this seems to work, I'm a bit unsure whether it's the best approach. Problem i that `getParent()` returns a `LazyFolder` and `method_exists($trashNode, 'getFileInfo')` three lines below returns `false`. Adding `getFileInfo()` to our own `LazyFolder` implementation (see above) doesn't change this - probably because the `LazyFolder` returned by `getParent()` is the implementation from server.

@juliushaertl and @blizzz maybe you have ideas/opinions?